src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -277,17 +273,32 @@ public OpenShiftOAuth2SecurityRealm(String serviceAccountDirectory, String servi
         else
         	populateDefaults();
     	
-        try {
-			hudson.util.PluginServletFilter.addFilter(new OpenShiftPermissionFilter());
-		} catch (ServletException e) {
-			LOGGER.log(Level.SEVERE, "ctor", e);
-		}
-        
         if (LOGGER.isLoggable(Level.FINE))
         	LOGGER.fine(String.format("ctor: derived default client id %s client secret %s sa dir %s transport %s", defaultedClientId, defaultedClientSecret, defaultedServiceAccountDirectory, transport));
     }
     
+    /*
+     * Note, a fair amount of investigation was done into leveraging the hudson.security.SecurityRealm.createFilter(FilterConfig) extension
+     * point.  However, quite a bit of tight coupling with assumed behavior in the jenkins core wrt the other security features and servlet filtering
+     * arose.  Hence, we are sticking with the hudson.util.PluginServletFilter.addFilter(Filter) path.
+     */
+    synchronized void createFilter() {
+        // restarts on things like plugin upgrade bypassed the call to the constructor, so filter initialization
+        // has to be driven in-line; note, after initial bring up, the filter variable will be set after subsequent
+        // jenkins restarts, but the addFilter call needs to be made on each restart, so we check flag to see if the filter
+        // has been ran through at least once
+        if (filter == null || !filter.initCalled) {
+            try {
+                filter = new OpenShiftPermissionFilter();
+                hudson.util.PluginServletFilter.addFilter(filter);
+            } catch (ServletException e) {
+                LOGGER.log(Level.SEVERE, "createFilter", e);
+            }
+        }
+    }
+    
     boolean populateDefaults() throws IOException, GeneralSecurityException {
+        createFilter();
```

Comment:
hmmmm ... its lined in mine and I thought my eclipse was converting everything to spaces.  I'll see what is up